Shopify/cli
# Switch to versioned app configuration beta flag

### WHY are these changes introduced?

Follow-up: https://github.com/Shopify/cli/pull/3250
Depends-on:
  - Partners: https://github.com/Shopify/partners/pull/52536

From the first PR regarding the n[ew versioned app configuration feature](https://github.com/Shopify/develop-app-foundations/issues/841) users could opt-in it using the env variable `SHOPIFY_CLI_VERSIONED_APP_CONFIG=1` in the `deploy` and `link` commands.
As it's required to opt-in for everyone at once, the CLI will switch from the use of the env variable to the use of the remote app `versioned app configuration` beta flag. This flag is used as well by the server side to enable the registration of the new `configuration` app modules


### WHAT is this pull request doing?
- Remove the support for the env variable `SHOPIFY_CLI_VERSIONED_APP_CONFIG` 
- Add support for the new app beta flag `versioned_app_config` returned by `partners`
- Swap the use of the env variable with the use of the beta flag. This means run a new API request to get the app betas where required.


### How to test your changes?
- Run the [same tests](https://github.com/Shopify/cli/pull/3161) suggested when the env variable `SHOPIFY_CLI_VERSIONED_APP_CONFIG` was added.
- Create first a new spin instance `spin up partners -n versioned-app-config-beta`
- Create a new remote app `NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=versioned-app-config-beta p shopify app config link --path /PATH/TO/YOUR/APP`
- This time after creating the remote app, the beta `versioned_app_config` should be enabled for the app from `internal partners dashboard`
- Run the commands against spin `NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=versioned-app-config-beta p shopify app deploy --path /PATH/TO/YOUR/APP`



### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
